rust/src/measurements/gumbel_max/mod.rs

```diff
@@ -24,6 +24,17 @@ pub enum Optimize {
     Min,
 }
 
+impl TryFrom<&str> for Optimize {
+    type Error = crate::error::Error;
+    fn try_from(s: &str) -> Fallible<Self> {
+        Ok(match s {
+            i if i.to_lowercase().starts_with("min") => Optimize::Min,
+            i if i.to_lowercase().starts_with("max") => Optimize::Max,
+            _ => return fallible!(FailedCast, "optimize must start with \"min\" or \"max"),
```

Comment:
A little concerned about parsing strings at this level. Is the `to_lowercase` essential? Do we need to support variations in case?

Reply:
I pulled string parsing just a bit further out by implementing Serialize and Deserialize. So strings aren't in the core, only in FFI.